gen/suites/actor_creation/illegal_actor.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+
+	"github.com/filecoin-project/test-vectors/chaos"
+	. "github.com/filecoin-project/test-vectors/gen/builders"
+)
+
+func actorAbortWithSystemExitCode(i int) func(v *Builder) {
+	return func(v *Builder) {
+		v.Messages.SetDefaults(GasLimit(1_000_000_000), GasPremium(1), GasFeeCap(200))
+
+		alice := v.Actors.Account(address.SECP256K1, abi.NewTokenAmount(1_000_000_000_000))
+		v.CommitPreconditions()
+
+		code := big.NewInt(int64(i))
+
+		msg := v.Messages.Raw(alice.ID, chaos.Address, chaos.MethodAbortWithSystemExitCode, MustSerialize(&code), Nonce(0), Value(big.Zero()))
+		v.CommitApplies()
+
+		v.Assert.Equal(exitcode.ExitCode(i), msg.Result.ExitCode)
+	}
+}
+
+func actorAbortWithSystemExitCodeSingle(v *Builder) {
+	v.Messages.SetDefaults(GasLimit(1_000_000_000), GasPremium(1), GasFeeCap(200))
+
+	alice := v.Actors.Account(address.SECP256K1, abi.NewTokenAmount(1_000_000_000_000))
+	v.CommitPreconditions()
+
+	var msgs []*ApplicableMessage
+	//for i := 0; i <= 15; i++ {
+	for i := 1; i <= 15; i++ {
+		code := big.NewInt(int64(i))
+
+		msg := v.Messages.Raw(alice.ID, chaos.Address, chaos.MethodAbortWithSystemExitCode, MustSerialize(&code), Nonce(uint64(i-1)), Value(big.Zero()))
+		msgs = append(msgs, msg)
+	}
+
+	v.CommitApplies()
+
+	v.Assert.Equal(exitcode.ExitCode(1), msgs[0].Result.ExitCode)
+	v.Assert.Equal(exitcode.ExitCode(2), msgs[1].Result.ExitCode)
+	v.Assert.Equal(exitcode.ExitCode(3), msgs[2].Result.ExitCode)
+	v.Assert.Equal(exitcode.ExitCode(4), msgs[3].Result.ExitCode)
+	v.Assert.Equal(exitcode.ExitCode(5), msgs[4].Result.ExitCode)
+	v.Assert.Equal(exitcode.ExitCode(6), msgs[5].Result.ExitCode)
+	v.Assert.Equal(exitcode.ExitCode(7), msgs[6].Result.ExitCode)
+	v.Assert.Equal(exitcode.ExitCode(8), msgs[7].Result.ExitCode)
+	v.Assert.Equal(exitcode.ExitCode(2), msgs[8].Result.ExitCode)
+	v.Assert.Equal(exitcode.ExitCode(2), msgs[9].Result.ExitCode)
+	v.Assert.Equal(exitcode.ExitCode(2), msgs[10].Result.ExitCode)
+	v.Assert.Equal(exitcode.ExitCode(2), msgs[11].Result.ExitCode)
+	v.Assert.Equal(exitcode.ExitCode(2), msgs[12].Result.ExitCode)
+	v.Assert.Equal(exitcode.ExitCode(2), msgs[13].Result.ExitCode)
```

Comment:
When running this variant of the test, I do not see WARNINGS from the VM for all messages >8, which is rather weird. I'd expect to see 15 warnings, but I see only 8. Maybe somehow related with the `nonce` again?

Reply:
I'm finding it a bit hard to see the mapping between the code the actor provides and the code we get here, but I think it's showing the VM doing the wrong thing. I think they all expect `SysErrorIllegalActor` (9).

`SysErrSenderStateInvalid` (2) suggests maybe something is wrong with the test code. Maybe you ran out of funds for gas?